builtin/providers/docker/data_source_docker_image.go

```diff
@@ -0,0 +1,78 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heroku/docker-registry-client/registry"
+)
+
+func dataSourceDockerImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDockerImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceDockerImageRead(d *schema.ResourceData, meta interface{}) error {
+	pullOpts := parseImageOptions(d.Get("name").(string))
+	auth := getAuthConfig(pullOpts)
```

Comment:
Would you mind moving this auth logic (basically from this line `32` to `57`) into provider as discussed in the other linked PR? Also the registry URL should be settable via provider field, e.g. `registry_url` or something along those lines.